### DISCUSSION TITLE
Extending Sentencizer to allow for "start of sentence chars" instead of just "punct chars"

### DISCUSSION THREAD

**Discussion by NixBiks** [explosion/spaCy#12683] [New Features & Project Ideas]
Currently you can only add punct chars as [input to the sentencizer](https://spacy.io/api/sentencizer#init).

It would be great to be able to also add characters that denotes the start of a sentence, e.g. a bullet point. I can make a PR if you think it's a good idea.

Let me know.

**Comment by adrianeboyd** [comment]
We appreciate the idea and can definitely see how it would be useful, but since it's relatively easy to write a custom rule-based component for sentence boundary detection, I think we'd prefer not to expand `sentencizer` with too many different possible settings.

**Reply by NixBiks** [reply]
I understand but adding another custom component for just that would require another iteration over all the tokens. Not very performant. 

I agree it's good to keep the component really simple. However this feature doesn't really increase the complexity IMO (very minimal). 

Are you certain you don't want to implement this?

**Reply by adrianeboyd** [reply]
To clarify a bit, I meant that it would be possible to write a custom component that handles both sentence-initial and sentence-final punctuation in the same component.

I will grant you that writing custom components in cython is a hassle and you would see speed differences in cython vs. python versions of the sentencizer (but I haven't done any detailed profiling recently).

**Reply by NixBiks** [reply]
Thanks for clarifying but yeah ;-)
